src/client/automation/playback/visible-element-automation.js

```diff
@@ -0,0 +1,120 @@
+import { delay, positionUtils, domUtils, arrayUtils, serviceUtils } from '../deps/testcafe-core';
+import getAutomationPoint from '../utils/get-automation-point';
+import screenPointToClient from '../utils/screen-point-to-client';
+import whilst from '../utils/promise-whilst';
+import { fromPoint as getElementFromPoint } from '../get-element';
+import AUTOMATION_ERROR_TYPES from '../errors';
+import AutomationSettings from '../settings';
+import MoveAutomation from './move';
+import { MoveOptions } from '../../../test-run/commands/options';
+
+
+export default class VisibleElementAutomation extends serviceUtils.EventEmitter {
+    constructor (element, offsetOptions) {
+        super();
+
+        this.WAITING_FOR_ELEMENT_STARTED_EVENT  = 'automation|waiting-element-started-event';
+        this.WAITING_FOR_ELEMENT_FINISHED_EVENT = 'automation|waiting-element-finished-event';
+
+        this.element            = element;
+        this.options            = offsetOptions;
+        this.automationSettings = new AutomationSettings(offsetOptions.speed);
+    }
+
+    _getElementForEvent (eventArgs) {
+        var { x, y }        = eventArgs.point;
```

Comment:
I agree that this should be named in the same way but they should be both with `_` since they are private methods